Update SCP image specification to v2.4

hadjilucasL opened the pull request:
Hi !

Just got my first Greaseweazle ! Found the code and wiki here invaluable to starting out archiving a lot of my old disk and would like to contribute something back to this effort. 

I am just beginning to get to grips with all the different formats but started backing things to SCP when I noticed the spec in [scp.py ](https://github.com/keirf/greaseweazle/blob/03e216e994bb67eec2864a9f70f0a802f6af0d30/src/greaseweazle/image/scp.py) is slightly out of date with respect to the [current version](https://www.cbmstuff.com/downloads/scp/scp_image_specs.txt). I believe the version implemented here is v2.2.

So made a small PR to introduce some hopefully non-breaking changes.
- Added AmigaHD as a disk type.
- Expanded the flag bits to include all 8 per v2.4 format

If this is heading along the right track I can expand this in another PR a bit more by adding the extension footer that is handy to store things like drive manufacturer, timestamp, creator (in this case Greaseweazle), comments and more. Currently I store this info in a spreadsheet as I am archiving this so might be useful to have it embedded in the image.

Can also contribute by porting the build system from setup.py fully to a pyproject.toml per PEP [518](https://peps.python.org/pep-0518/)/ [621](https://peps.python.org/pep-0621/).

Hope it helps! Kind regards and many thanks for this project, Lucas :blush:

hadjilucasL commented on src/greaseweazle/image/scp.py:
```diff
@@ -5,50 +5,65 @@
-    'hdd-rll':     0xf1
+    'c64':          0x00,
+    'amiga':        0x04,
+    'amigahd':      0x08,
```
Looks like a lot of changes but I just 
- added the `amigahd` 
- and changed the case of `appleII` and `appleIIpro` to all lowercase as the disktype.setter method lowers everything before the match

keirf commented on src/greaseweazle/image/scp.py:
```diff
@@ -50,6 +51,16 @@
 
 
+class SCPHeaderFlags(IntFlag):
+    INDEXED = auto()        # if set, the image used the index mark to queue tracks
```
I don't agree with using `auto()` here. Value assignments are not guaranteed by the library API. Spell out the assignments. `INDEXED = 1` and so on.

keirf replied:
Ok, maybe the assigned values are guaranteed by the API. But I still don't agree with use of `auto()` when specific values are required. So switch to `1, 2, 4, ...` or `1<<0, 1<<1, 1<<2, ...`

hadjilucasL replied:
Yeap, I agree. Better to be explicit on the values. Fixed :+1:

keirf commented on src/greaseweazle/image/scp.py:
```diff
@@ -110,7 +121,7 @@ def from_file(cls, name):
 
         header = struct.unpack("<3s9BI", dat[0:16])
-        sig, _, disk_type, nr_revs, _, _, flags, _, single_sided, _, _ = header
+        sig, version, disk_type, nr_revs, _, _, flags, _, single_sided, _, _ = header
```
Why?

hadjilucasL replied:
Ah I reverted this later - will remove the patch for this.

keirf commented on src/greaseweazle/image/scp.py:
```diff
@@ -14,38 +14,39 @@
-    'hdd-rll':     0xf1
+    'c64':          0x00,
+    'amiga':        0x04,
+    'disk_AmigaHD': 0x08,
```
Should be just `AmigaHD` as we do not include `disk_` prefix on any other member. This will also avoid reformatting all the other lines and make the diff smaller. :)

hadjilucasL replied:
ah yes, the extra length pushed all the values +1 column. Done :+1:

keirf commented on src/greaseweazle/image/scp.py:
```diff
@@ -54,6 +54,7 @@
 
 
 class SCPHeaderFlags(IntFlag):
+    # Warning - Order of flags here should align with SCP specification
```
This won't be necessary when you get rid of `auto()`.

keirf commented:
Thanks, I have reviewed the commits.

Regarding adding a footer, it sounds good. One concern I have is what we will test against to be sure we are producing good SCP images. Testing greaseweazle against itself won't be good enough. 

Regarding pyproject.toml, also good if you know what you are doing with Python packaging. It took some pain to get the greaseweazle setup to where it is today. I wonder if there is an equivalent to setuptools_scm in pyproject.toml?

hadjilucasL commented:
Thanks for the helpful comments ! I addressed the feedback and rewrote the patches to be better aligned with the intended changes. There were a lot of unnecessary commits in the original set.

Re the footer mm I will think about it. Would you say a good test is to check if the image loads in the current [SuperCard Pro Software](https://www.cbmstuff.com/index.php?route=product/product&product_id=52?) ? I am kinda hoping it is updated to read and display the footer info but will need to check. Not sure if any other tool such as HxC does this.

On the pyproject toml, I will take a stab at it this weekend. I had a quick look and there is guidance on using [setuptools_scm  with it](https://github.com/pypa/setuptools_scm#pyprojecttoml-usage) so the git versioning aspect should be maintained.

hadjilucasL commented:
Will do. Thanks Keir !
